src/Digdir.Domain.Dialogporten.Infrastructure/Altinn/OrganizationRegistry/OrganizationRegistryClient.cs

```diff
@@ -28,6 +30,35 @@ public OrganizationRegistryClient(HttpClient client, IFusionCacheProvider cacheP
         return orgShortName;
     }
 
+    public async Task<OrganizationLongName[]> GetOrganizationLongNames(string orgNumber, CancellationToken cancellationToken)
+    {
+        var orgLongNamesByOrgNumber = await _cache.GetOrSetAsync(OrgLongNameReferenceCacheKey, async token => await GetOrgLongNamesByOrgNumber(token), token: cancellationToken);
+        orgLongNamesByOrgNumber.TryGetValue(orgNumber, out var orgLongNames);
+
+        return orgLongNames?.LongNames.ToArray() ?? Array.Empty<OrganizationLongName>();
+    }
+
+    private async Task<Dictionary<string, OrganizationLongNames>> GetOrgLongNamesByOrgNumber(CancellationToken cancellationToken)
```

Comment:
after talk with @oskogstad , decided to refactor this into a getOrgInfoByOrgNumber and combining these values to avoid 2 different cache keys

Reply:
Refactored